summingbird-online/src/main/scala/com/twitter/summingbird/online/executor/AsyncBase.scala

```diff
@@ -34,6 +35,30 @@ object AsyncBase {
    * finished.
    */
   val OutstandingFuturesDequeueRatio = 2
+
+  /**
+   * Wait for n futures to finish. Doesn't block, the returned future is satisfied
+   * once n futures have finished either successfully or unsuccessfully.
+   * If n is greater than number of futures in queue then we wait on all of them.
+   */
+  def waitN[A](fs: Queue[Future[A]], n: Int): Future[Unit] = {
+    require(n >= 0)
+    val waitOnCount = Math.min(fs.size, n)
+    if (waitOnCount == 0) {
+      Future.Unit
+    } else {
+      val count = new AtomicInteger(waitOnCount)
+      val p = Promise[Unit]()
+      for (f <- fs) {
+        f ensure {
+          if (count.decrementAndGet() == 0) {
+            p.setValue(())
```

Comment:
can we add a comment here, something like:
```
Note, since we are only decrementing, we can only cross the value 0 one time (unless we decrement > 2^(32) times).
```